### ISSUE TITLE
Remove the in-game editor

### ISSUE THREAD

**Issue by samjar** [ironbane/IronbaneServerLegacy#176]
Make it possible to add all the models (trees, rocks, etc.), movable and toggable platforms, chests, levers, maybe even NPC's and nodes straight from Blender. While the in-game editor is awesome in its own ways, and gives room for collaboration, being able to add all that through Blender would give me much more flexibility.

[8/23/2013 1:15:39 PM] Ben Sparks: so I was thinking about the houses and decorations and whatnot, I realize live editing is neat, but maybe we should place all that stuff in the model editor so that it lines up right instead. just leave the npcs and such as dynamic
[8/23/2013 1:16:51 PM] Sam Jarva: yeah, I agree with you. I asked Nick a while back if I could at least place the houses, which are rather large models, straight into the .obj
[8/23/2013 1:17:18 PM] Sam Jarva: However, placing the 1000's of trees will probably make the file too heavy
[8/23/2013 1:17:21 PM] Ben Sparks: we'd lose the ability to have multiple people work on it at the same time, but that's not a super pressing need IMO
[8/23/2013 1:17:56 PM] Ben Sparks: yeah the trees is an issue certainly
[8/23/2013 1:18:20 PM] Sam Jarva: I'm not sure where the invisible threshold for how large the .js file can be before it slows everything done considerably. Maybe we should just stick to larger models
[8/23/2013 1:18:31 PM] Sam Jarva: It always annoyed me how you couldn't see the village from far away, anyway lol
**[8/23/2013 1:18:32 PM] Ben Sparks: tho if we created special objects and just placed cubes with a meta tag that the game could replace with trees that might be better**

Your thoughts?

**Comment by nickjanssen** [comment]
I did think more about this. There are a couple of things to be considered as well:
- It would be technically very difficult and take a lot of time to make Blender place game entities like monsters and NPC's. Even if we would manage to represent game models using points in Blender, you would not be able to see the bounds of the mesh and it will still be a guess where the model would be on the terrain mesh.
- I know that Sam wants to place models so they can be seen from a distance, but what if there was an option to toggle each 3d model so they can be seen from far away, regardless of the players position in that zone?
- What if we made placing models in-game less buggy and allow you to move things in a Blender fashion? And implement a similar way of rotating/moving around as in Blender?
- For things that are suddenly floating in the air or sunk in the ground because you changed the terrain, we can add a function for 3d models to automatically lower themselves to the ground.
- I think it is also better if Sam focuses on terrain only. Anything that is a small part should become an 3d model. For instance stairs, small walls, small rocky objects which are currently present in the tutorial mesh should become a separate 3d model. This way we all have more control over the result in-game. This is the same way WoW functions, they only have a terrain mesh where you can place models on.

In short, I suggest that we focus on making a better in-game editor and leave Blender ONLY for the terrain. If whatever you are adding can be considered an object, make it a separate 3d model that can be placed individually. This includes churches, inns, large boulders, trees, etc. With the combination of the above points I think we can turn our current editor much more powerful and versatile.

**Comment by bsparks** [comment]
long distance models usually have several LoD versions, the engine would need to support that, but it would help.

I think that if we do improve the in-game editor that it should be a mouse lock mode (chrome only?) supported thing. I was already thinking before about various ways to improve it.

I still think that it might be faster / better to place all of the buildings and static level stuff in blender. The NPCs and dynamic things was always more questionable to me. But the in-game editor _can_ do the same stuff, my thoughts were that Blender has spent a lot more time refining it already.

Another sort of related note, why don't we use the terrain mesh again? The problem before was that it was tiled in small chunks, would having a larger terrain mesh like the current world size be an issue? It's the same thing really isn't it poly wise? I think that if we could manipulate the terrain in the editor it would be easier to line things up.
